File: QytetetJava/src/modeloqytetet/src/modeloqytetet/Sorpresa.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package modeloqytetet;

/**
 *
 * @author joseng2709
 */
public class Sorpresa {
    private String texto;
    private TipoSorpresa tipo;
    private int valor;
    
    public Sorpresa (String texto ,int valor ,TipoSorpresa tipo){
        this.texto=texto;
        this.valor=valor;
        this.tipo=tipo;
    }
    
    int getValor(){
        return valor;
    }
    
    TipoSorpresa getTipoSorpresa(){
        return tipo;
    }
    
    String getTexto(){
        return texto;
    }
    
    @Override
    public String toString() {
        return "Sorpresa{" + "texto=" + texto + ", valor=" + 
        Integer.toString(valor) + ", tipo=" + tipo + "}"+"\n";
    } 
    
    
}

File: Qytetet_Ruby/pkg/Qytetet_Personal-0.0.1/lib/sorpresa.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

class Sorpresa
  attr_accessor:texto
  attr_accessor:tipo
  attr_accessor:valor
  def initialize(unString,unTipoSorpresa,unInt)
    @texto=unString
    @tipo=unTipoSorpresa
    @valor=unInt
  end
  
  @Overrride
  def to_s()
    return "Texto: #{@texto}\n"\
           "Valor: #{@valor}\n"\
           "Tipo: #{@tipo}\n\n"
  end
  
  def get_texto()
    return @texto
  end
  
  def get_tipo()
    return @tipo
  end
  
  def get_valor()
    return @valor
  end
end
